### ISSUE TITLE
Include achievement progess in the crafting calculator

### ISSUE THREAD

**Issue by queicherius** [gw2efficiency/issues#104]
For achievements, the calculator should take a look at the specific achievement bits to figure out if parts of the achievements (that don't reward items) are already completed.

**Comment by Azania** [comment]
It is very annoying to see the calculator telling you need more materials for no reason. Also prime example of this is the precursor collection 2. Which is all crafting/gathering based and doesn't give you an item in return.

The ones you get from crafting into the collection do count but only while you keep them in your inventory somewhere without selling/destroying.

**Comment by queicherius** [comment]
> Also prime example of this is the precursor collection 2.

Yes, I was thinking mainly about the precursor crafting too. Are there any more cases where achievements get used so heavily?

**Comment by coldwaterq** [comment]
The legendary backpack, but that is sort of a precursor as well.

**Comment by Azania** [comment]
I will go over a basic list later but basically _every_ achievement under the collections category has the potential to have stuff for crafting.

**Comment by queicherius** [comment]
I just asked Mystic, who does all the recipe stuff, if he can include the achievement id into recipes that are produced by achievements. That would make things pretty easy. :)

**Comment by queicherius** [comment]
Recipes now include their `achievement_id`.

- [ ] Update `custom-recipes` to support additional fields (and while I am at it rewrite it into a JS module)
- [ ] Update `recipe-nesting` to support additional fields
- [ ] Update `gw2efficiency` to support the achievement ids, loading the current status and checking them as owned in the progress

**Comment by Azania** [comment]
Oki will wait for it to come online then. Atm not working heh :) Sorry for the long silence.. if needed I can check later on full things going on in categories; short list for now:

- Items gained in collection only:
- Items from vendor
- Physical items from a recipe
- Physical items from achievement progress
- Items gained in collection from another collection.
- Chained above in recipes/gathering/achievement collections..

You can destroy/sell some items from the above.

Nevermore 4 seems a perfect example of all the above.. Same with other several collections of other next gen leggies.

I hope the `achievement_id `is also the id of an invidual part of the collection. If so it should work from what you just said.

**Comment by queicherius** [comment]
This is deployed and *should* be working now!

**Comment by Azania** [comment]
Hard to test when Chuka and Champawat II shows dreamer collection 2 instead of the chuka one. The amount of gold needed is also wrong due to those differences. I will also make a reddit post seeing this one is a separate issue.

**Comment by Azania** [comment]
On topic: working mostly, seems to ignore skin items. I just checked Wild Abadon, it does not see my Plated Torch yet I have other parts of that collection. Should not be possible, if needed I can supply screenshots. Guess I forgot to include skins into my list above with 'Physical' items that include a skin heh.

So I would open this again or make a new issue for skin related items.

**Comment by queicherius** [comment]
Hey! I forwarded the bug with the Chunka collection.

> Is this not auto read from the API but set manually?

Yes, all these recipes are [manually managed](https://github.com/gw2efficiency/custom-recipes), because they are not in the API. There is *some* automation with the achievement API, but it doesn't include Mystic Forge recipes, vendors etc, so there always has to be manual work.

I opened another issue with the skin items: #142
